### PR TITLE
infra: Golang coverage fixes

### PR DESCRIPTION
**Pull request by catenacyber**
@AdamKorcz what do you think of these commits ?

cf https://github.com/google/oss-fuzz/pull/8937 and should fix https://github.com/google/oss-fuzz/issues/9808

### CODE REVIEW

File: infra/base-images/base-runner/coverage [hunk 1]
```diff
@@ -137,10 +137,10 @@ function run_go_fuzz_target {
   # setup for native go fuzzers
   cd $OUT
   mkdir -p "testdata/fuzz/${target}"
-  cp -r "${FUZZ_CORPUS_DIR}*" "testdata/fuzz/${target}/"
+  cp -r "${FUZZ_CORPUS_DIR}" "testdata/fuzz/"
```
Comment (catenacyber) [inline infra/base-images/base-runner/coverage:140]: This looks like a true bug looking previously for the file named `*`

File: infra/base-images/base-runner/coverage [hunk 2]
```diff
@@ -137,10 +137,10 @@ function run_go_fuzz_target {
...
 
-  # rewrite libFuzzer corpus to Std Go corpus
-  $SYSGOPATH/bin/convertcorpus $target "testdata/fuzz/${target}"
+  # rewrite libFuzzer corpus to Std Go corpus if native fuzzing
+  grep "TestFuzzCorpus" $target > /dev/null 2>&1 && $SYSGOPATH/bin/convertcorpus $target "testdata/fuzz/${target}"
```
[the remark this answers is not in the document]
Comment (AdamKorcz) [reply infra/base-images/base-runner/coverage:143]: Yes, that is fine.

**Review by AdamKorcz — APPROVED** [review]
LGTM

**Comment by catenacyber** [comment]
For reference, tested with
```
python3 infra/helper.py build_image --no-pull base-runner
python3 infra/helper.py build_image --no-pull base-builder-go
python3 infra/helper.py build_image --no-pull ngolo-fuzzing
python3 infra/helper.py build_fuzzers --sanitizer coverage ngolo-fuzzing
python3 infra/helper.py coverage --no-corpus-download ngolo-fuzzing
```

**Comment by catenacyber** [comment]
@Alan32Liu could you run gcbrun please ?

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer coverage --fuzzing-engine ngolo-fuzzing

**Comment by catenacyber** [comment]
Thanks, gcbrun is happy :-)

**Pull request merged**